pkg/agent/util.go

```diff
@@ -22,11 +25,26 @@ func getTimestamp() v1Time.Time {
 	return newV1Time
 }
 
-// ApplyResouceToConfig - applies the resources to agent configs
+// ApplyResourceToConfig - applies the resources to agent configs
 // Uses reflection to get the IResourceConfigCallback interface on the config struct or
 // struct variable.
 // Makes call to ApplyResources method with dataplane and agent resources from API server
-func ApplyResouceToConfig(cfg interface{}) error {
+func ApplyResourceToConfig(cfg interface{}) error {
+	obj := cfg
+
+	// This defer func is to catch a possible panic that WILL occur if the cfg object that is passed in embedds the IResourceConfig interface
+	// within its struct, but does NOT implement the ApplyResources method. While it might be that this method really isn't necessary, we will
+	// log an error alerting the user in case it wasn't intentional.
+	defer func() {
+		if err := recover(); err != nil {
+			str := fmt.Sprintf("%v", err)
+			if strings.Contains(str, "nil pointer dereference") {
+				log.Errorf("The function 'ApplyResources' for interface IResourceConfigCallback is not implemented in %s.", reflect.TypeOf(obj))
```

Comment:
Should this be an error and a message specific for devs or a warn and a message like "ApplyResources not yet implemnted"? Maybe ask Cyrille?

Reply:
I'll ask

Reply:
C says to do a warn. I'll also check the SDK docs to be sure that implementing this is documented.